mcp/server.go

```diff
@@ -240,8 +240,12 @@ func toolForErr[In, Out any](t *Tool, h ToolHandlerFor[In, Out]) (*Tool, ToolHan
 			}, nil
 		}
 
-		// TODO(v0.3.0): Validate out.
-		_ = outputResolved
+		// Validate output schema, if any.
+		// Skip if out is nil: we've removed "null" from the output schema, so nil won't validate.
+		if v := reflect.ValueOf(out); v.Kind() == reflect.Pointer && v.IsNil() {
```

Comment:
But `reflect.ValueOf(untyped nil)` is invalid, no? Should you also add a v.IsValid() check here?
In that case, I'm concerned as to why our tests pass.

Reply:
You can call v.Kind on an invalid reflect.Value, which is all we do with it. (It will fail the reflect.Pointer check and we'll never call IsNil).